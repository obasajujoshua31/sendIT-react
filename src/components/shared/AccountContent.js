import React, {Component} from "react";
import { withRouter} from 'react-router-dom';

class AccountContent extends Component{
  constructor(props){
    super(props)
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick (){
    this.props.history.push('/signup');
  }
  render(){

  
  return (
    <React.Fragment>
      <div className="service-information">
        <h1><span>Just Call </span>(0908754647846)</h1>
        <p>
          We are always ready for your service
        </p>
        <button onClick ={this.handleClick}>SendIT Today</button>
      </div>
    </React.Fragment>
  );
  }
};

export default withRouter(AccountContent);
