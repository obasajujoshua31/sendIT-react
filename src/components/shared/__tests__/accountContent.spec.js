import React from "react";
import sinon from "sinon";
import { shallow } from "enzyme";
import { AccountContent, mapDispatchToProps } from "../AccountContent";

const props = {
  loadParcels: jest.fn(),
  history: {
    push: jest.fn()
  }
};

describe("AccountContent", () => {
  const wrapper = shallow(<AccountContent {...props} />);
  it("should render Account Content component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should contain 1 <div> tag", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });
  it("should contain 1 <h1> tag", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });
  it("should contain 1 <p> tag", () => {
    expect(wrapper.find("p").length).toEqual(1);
  });
  it("should contain 1 <button> tag", () => {
    expect(wrapper.find("button").length).toEqual(1);
  });
  it("should call load parcels", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
  });
});
