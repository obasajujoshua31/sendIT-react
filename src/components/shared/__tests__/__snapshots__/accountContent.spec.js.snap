// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountContent should render Account Content component 1`] = `
<Fragment>
  <div
    className="service-information"
  >
    <h1>
      <span>
        Just Call 
      </span>
      (0908754647846)
    </h1>
    <p>
      We are always ready for your service
    </p>
    <button
      onClick={[Function]}
    >
      SendIT Today
    </button>
  </div>
</Fragment>
`;
